src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantACustomTFM.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using FluentAssertions;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Commands;
+using Xunit;
+using Xunit.Abstractions;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+
+namespace Microsoft.NET.Build.Tests
+{
+    public class GivenThatWeWantACustomTFM : SdkTest
+    {
+        public GivenThatWeWantACustomTFM(ITestOutputHelper log) : base(log)
+        {}
+
+        [Fact]
+        public void It_imports_custom_parsing_targets()
+        {
+            TestProject testProject = new TestProject()
+            {
+                Name = "CustomTFMProject",
+                IsSdkProject = true, 
+                IsExe = true, 
+                TargetFrameworks = "netcoreapp3.0"
+            };
+
+            testProject.AdditionalProperties["CustomTargetFrameworkParsingTargets"] = @"$(MSBuildProjectDirectory)\CustomTargetFramework.targets";
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+
+            var propertyName = "CustomTargetFrameworkProp";
+            File.WriteAllText(Path.Combine(testAsset.TestRoot, testProject.Name, "CustomTargetFramework.targets"), $@"
+<Project ToolsVersion=`14.0` xmlns=`http://schemas.microsoft.com/developer/msbuild/2003`>
+  <PropertyGroup>
+    <{ propertyName }>TargetFramework: $(TargetFramework), TargetFrameworkVersion: $(TargetFrameworkVersion)</{ propertyName }>
```

Comment:
I would suggest trying to make this test match the scenario this feature was requested for more closely.  For example:

- In the project file, set the `TargetFramework` to `netcoreapp3.0-<RuntimeIdentifier>`
- In the custom .props, parse out the portions of the `TargetFramework` into `TargetFramework` and `RuntimeIdentifier`
- Let the default `TargetFramework` parsing logic in the SDK parse out the "stripped" `TargetFramework`
- Verify that `TargetFramework`, `TargetFrameworkIdentifier`, `TargetFrameworkVersion`, and `RuntimeIdentifier` properties are all set correctly.

There were other hoops the runtime team had to jump through to get everything to work, so it's possible that such a test won't work.  But I think the problems would be if you tried to restore and build, rather than just get the property values.